src/eth/storage/postgres/postgres.rs

```diff
@@ -56,7 +56,8 @@ impl EthStorage for Postgres {
                     balance as "balance: _",
                     bytecode as "bytecode: _"
                 FROM accounts
-                WHERE address = $1 AND block_number = $2
+                WHERE address = $1 AND block_number <= $2
+                ORDER BY block_number DESC
```

Comment:
can you add some comments explaining why we need to order by number here? just for the record

Reply:
Sure